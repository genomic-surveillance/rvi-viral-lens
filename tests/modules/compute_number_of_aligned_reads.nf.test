nextflow_process {

    name "Test Process compute_number_of_aligned_reads"
    script "modules/compute_number_of_aligned_reads.nf"
    process "compute_number_of_aligned_reads"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    id:"test.9000000",
                    sample_id:"test",
                    taxid:9000000
                ]
                test_loc="${projectDir}/tests/test_data"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    params.meta,
                    file("${params.test_loc}/test_alignments/test2.9000000.sorted.bam")
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
