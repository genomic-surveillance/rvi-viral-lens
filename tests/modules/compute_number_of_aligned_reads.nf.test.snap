{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        "45225|15|\n"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-02-29T10:22:23.126122"
    }
}